src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,126 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Container class to hold a list of tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    public ArrayList<Task> getList() {
+        return tasks;
+    }
+
+    /**
+     * Returns the list of tasks that matches the specified date. <br>
+     * by performing date comparison based on the specified type.<br><br>
+     *
+     * comparisonType is defined as follows:
+     * <li>0 : Tasks happening on the specified date. </li>
+     * <li>1 : Tasks happening before the specified date.</li>
+     * <li>2 : Tasks happening after the specified date.</li>
+     *
+     * @param date The date that is to be compared with.
+     * @param comparisonType The type of comparison to be performed.
+     * @return A list of tasks that matches the specified date and comparisonType.
+     */
+    public ArrayList<Task> getTasksBasedOnDate(LocalDate date, int comparisonType) {
```

Comment:
Does `getTasksByDate` convey the same meaning more concisely?
```suggestion
    public ArrayList<Task> getTasksByDate(LocalDate date, int comparisonType) {
```